File: benchmarks/pandas/bench_read_json.py
"""
Benchmark: DataFrame read_json

Parses a JSON string into a DataFrame (records orient).
Outputs JSON: {"function": "read_json", "mean_ms": ..., "iterations": ..., "total_ms": ...}
"""

import json
import time
import io

import pandas as pd

ROWS = 5_000
WARMUP = 5
ITERATIONS = 50

records = [
    {"id": i, "x": i * 1.1, "y": i * 2.2, "label": f"item_{i % 100}"}
    for i in range(ROWS)
]
json_str = json.dumps(records)

for _ in range(WARMUP):
    pd.read_json(io.StringIO(json_str))

times: "list[float]" = []
for _ in range(ITERATIONS):
    start = time.perf_counter()
    pd.read_json(io.StringIO(json_str))
    end = time.perf_counter()
    times.append((end - start) * 1000)

total_ms = sum(times)
mean_ms = total_ms / ITERATIONS

print(json.dumps({
    "function": "read_json",
    "mean_ms": round(mean_ms, 3),
    "iterations": ITERATIONS,
    "total_ms": round(total_ms, 3),
}))

File: benchmarks/tsb/bench_read_json.ts
/**
 * Benchmark: DataFrame readJson
 *
 * Parses a JSON string into a DataFrame (records orient).
 * Outputs JSON: {"function": "read_json", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */

import { readJson } from "../../src/index.ts";

const ROWS = 5_000;
const WARMUP = 5;
const ITERATIONS = 50;

function makeJsonString(): string {
  const records = Array.from({ length: ROWS }, (_, i) => ({
    id: i,
    x: i * 1.1,
    y: i * 2.2,
    label: `item_${i % 100}`,
  }));
  return JSON.stringify(records);
}

const jsonStr = makeJsonString();

for (let i = 0; i < WARMUP; i++) {
  readJson(jsonStr);
}

const times: number[] = [];
for (let i = 0; i < ITERATIONS; i++) {
  const start = performance.now();
  readJson(jsonStr);
  const end = performance.now();
  times.push(end - start);
}

const totalMs = times.reduce((a, b) => a + b, 0);
const meanMs = totalMs / ITERATIONS;

console.log(JSON.stringify({
  function: "read_json",
  mean_ms: Math.round(meanMs * 1000) / 1000,
  iterations: ITERATIONS,
  total_ms: Math.round(totalMs * 1000) / 1000,
}));
